biocrnpyler/parameter.py

```diff
@@ -64,103 +45,319 @@
+            raise ValueError(f"parameter_name must be a string: received {type(new_parameter_name)}.")
+        if not re.search('^[a-z]+', new_parameter_name, re.IGNORECASE):
+            raise ValueError(f'parameter_name should be at least one character and cannot start with a number!')
+
+        self._parameter_name = new_parameter_name
+
+    @property
+    def value(self) -> numbers.Real:
+        return self._value
+
+    @value.setter
+    def value(self, new_parameter_value: Union[str, numbers.Real]):
+        if not (isinstance(new_parameter_value, numbers.Real) or isinstance(new_parameter_value, str)):
+            raise ValueError(f"parameter_value must be a float or int: received {type(new_parameter_value)}.")
+        if isinstance(new_parameter_value, str):
+            if re.search('[a-d-f-z]', new_parameter_value, re.I) \
+                    or re.search('(^[1-9]+/[1-9]+)|(^[1-9]+e-?[0-9]+)|(^.?[0-9])', new_parameter_value, re.I) is None:
+                raise ValueError(f'No valid parameter value! Accepted formats: 1.00 or 1e4 or 2/5, we got {new_parameter_value} ')
+
+            self._value = Parameter._convert_rational(new_parameter_value)
         else:
-            raise ValueError("can't parse value of parameter %s" % name)
+            self._value = new_parameter_value
 
     @staticmethod
-    def _get_field_names(field_names: List[str], accepted_field_names: Dict[str, List[str]]) -> Dict[str, str]:
-        """ Searches through valid field names and finds the currently used one. It builds a dictionary of currently
-            used field names
-        :param field_names: list of field names (columns) found in the csv file
-        :param accepted_field_names: dictionary of possible field names and their valid aliases
-        :return: dictionary of currently used field names (aliases)
+    def _convert_rational(p_value: str) -> numbers.Real:
+        if '/' in p_value:
+            nom, denom = p_value.split('/')
+            return float(nom)/float(denom)
+        else:
+            return float(p_value)
+
+
+class ParameterEntry(Parameter):
+    """
+    A class for representing parameters in a parameter stored the ParameterDatabase
+     parameter_keys is a dictionary {key:value} or named_tuple (type ParameterKey) of keys for looking up the parameter
+     parameter_info is a dictionary {key:value} of additional information about the parameter. 
+         For example: additional columns in the parameter file or the parameter file name.
+    """
+    def __init__(self, parameter_name: str, parameter_value: Union[str,numbers.Real], parameter_key=None, parameter_info=None):
+        Parameter.__init__(self, parameter_name, parameter_value)
+
+        self.parameter_key = parameter_key
+        self.parameter_info = parameter_info
+
+    #Helper function to create ParameterKeys
+    @staticmethod
+    def create_parameter_key(new_key: Union[Dict, ParameterKey, str], parameter_name=None) -> ParameterKey:
```

Comment:
should be this method private? i.e adding an underscore at the beginning of the method?
```python
def _create_parameter_key(new_key: Union[Dict, ParameterKey, str], parameter_name=None) -> ParameterKey:
```

Reply:
I could imagine someone wanting to use the method - for example to convert parameter values in a dictionary or another program to biocrnpyler parameters.